From 5be3ffdf767c1efcbfd2d1be87aa83f2e37e348e Mon Sep 17 00:00:00 2001
From: Mingli Yu <mingli.yu@windriver.com>
Date: Mon, 1 Aug 2022 15:44:38 +0800
Subject: [PATCH 4/5] config_info.c: not expose build info

Don't collect the build information to fix the buildpaths issue.

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Mingli Yu <mingli.yu@windriver.com>
---
 configure.ac             |  2 +-
 src/common/config_info.c | 70 +---------------------------------------
 2 files changed, 2 insertions(+), 70 deletions(-)

diff --git a/configure.ac b/configure.ac
index 27f382d..3dd6bb1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -23,7 +23,7 @@ AC_COPYRIGHT([Copyright (c) 1996-2023, PostgreSQL Global Development Group])
 AC_CONFIG_SRCDIR([src/backend/access/common/heaptuple.c])
 AC_CONFIG_AUX_DIR(config)
 AC_PREFIX_DEFAULT(/usr/local/pgsql)
-AC_DEFINE_UNQUOTED(CONFIGURE_ARGS, ["$ac_configure_args"], [Saved arguments from configure])
+AC_DEFINE_UNQUOTED(CONFIGURE_ARGS, ["ac_configure_args"], [Saved arguments from configure])
 
 [PG_MAJORVERSION=`expr "$PACKAGE_VERSION" : '\([0-9][0-9]*\)'`]
 [PG_MINORVERSION=`expr "$PACKAGE_VERSION" : '.*\.\([0-9][0-9]*\)'`]
diff --git a/src/common/config_info.c b/src/common/config_info.c
index 09e78a6..86e4230 100644
--- a/src/common/config_info.c
+++ b/src/common/config_info.c
@@ -38,7 +38,7 @@ get_configdata(const char *my_exec_path, size_t *configdata_len)
 	int			i = 0;
 
 	/* Adjust this to match the number of items filled below */
-	*configdata_len = 23;
+	*configdata_len = 14;
 	configdata = palloc_array(ConfigData, *configdata_len);
 
 	configdata[i].name = pstrdup("BINDIR");
@@ -123,74 +123,6 @@ get_configdata(const char *my_exec_path, size_t *configdata_len)
 	configdata[i].setting = pstrdup(path);
 	i++;
 
-	configdata[i].name = pstrdup("CONFIGURE");
-	configdata[i].setting = pstrdup(CONFIGURE_ARGS);
-	i++;
-
-	configdata[i].name = pstrdup("CC");
-#ifdef VAL_CC
-	configdata[i].setting = pstrdup(VAL_CC);
-#else
-	configdata[i].setting = pstrdup(_("not recorded"));
-#endif
-	i++;
-
-	configdata[i].name = pstrdup("CPPFLAGS");
-#ifdef VAL_CPPFLAGS
-	configdata[i].setting = pstrdup(VAL_CPPFLAGS);
-#else
-	configdata[i].setting = pstrdup(_("not recorded"));
-#endif
-	i++;
-
-	configdata[i].name = pstrdup("CFLAGS");
-#ifdef VAL_CFLAGS
-	configdata[i].setting = pstrdup(VAL_CFLAGS);
-#else
-	configdata[i].setting = pstrdup(_("not recorded"));
-#endif
-	i++;
-
-	configdata[i].name = pstrdup("CFLAGS_SL");
-#ifdef VAL_CFLAGS_SL
-	configdata[i].setting = pstrdup(VAL_CFLAGS_SL);
-#else
-	configdata[i].setting = pstrdup(_("not recorded"));
-#endif
-	i++;
-
-	configdata[i].name = pstrdup("LDFLAGS");
-#ifdef VAL_LDFLAGS
-	configdata[i].setting = pstrdup(VAL_LDFLAGS);
-#else
-	configdata[i].setting = pstrdup(_("not recorded"));
-#endif
-	i++;
-
-	configdata[i].name = pstrdup("LDFLAGS_EX");
-#ifdef VAL_LDFLAGS_EX
-	configdata[i].setting = pstrdup(VAL_LDFLAGS_EX);
-#else
-	configdata[i].setting = pstrdup(_("not recorded"));
-#endif
-	i++;
-
-	configdata[i].name = pstrdup("LDFLAGS_SL");
-#ifdef VAL_LDFLAGS_SL
-	configdata[i].setting = pstrdup(VAL_LDFLAGS_SL);
-#else
-	configdata[i].setting = pstrdup(_("not recorded"));
-#endif
-	i++;
-
-	configdata[i].name = pstrdup("LIBS");
-#ifdef VAL_LIBS
-	configdata[i].setting = pstrdup(VAL_LIBS);
-#else
-	configdata[i].setting = pstrdup(_("not recorded"));
-#endif
-	i++;
-
 	configdata[i].name = pstrdup("VERSION");
 	configdata[i].setting = pstrdup("PostgreSQL " PG_VERSION);
 	i++;
-- 
2.25.1

